src/common/error/src/ext.rs

```diff
@@ -34,12 +34,37 @@ pub trait ErrorExt: std::error::Error + StackError {
     /// Returns the error as [Any](std::any::Any) so that it can be
     /// downcast to a specific implementation.
     fn as_any(&self) -> &dyn Any;
+
+    fn output_msg(&self) -> String
+    where
+        Self: Sized,
+    {
+        let error = self.last();
+        if let Some(external_error) = error.source() {
+            format!("{error}: {external_error}")
+        } else {
+            format!("{error}")
+        }
+    }
```

Comment:
As we are using `err.to_string()` for error output, is it possible to overwrite `to_string` directly so that developers can use `to_string` as normal.

It can be something like:

```
impl<E> fmt::Display for E where E: ErrroExt {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "SuperError is here!")
    }
}


```

Reply:
`Display` is occupied by snafu, we have to implement it on another fn